Fixes additional flags in srun_fastsurfer and T2 processing

dkuegler opened the pull request:
Additional flags (including --t2) did not work properly from the subject_list file in srun_fastsurfer.sh 
The Bug is actually in stools.sh:translate_cases, where the lines are not correct parsed. 
This PR fixes this issue and properly parses the file. Also, the check to find non-existent files in stools.sh:check_subject_images.

This PR also adds a check if files are symlinks, which may cause problems with folders mounted in singularity (this needed a small change in srun_fastsurfer.sh - an additional argument to check_subject_images).

This PR fixes #654

dkuegler committed:
Fixes additional flags in srun_fastsurfer

Additional flags (including --t2) did not work properly from the subject_list file in srun_fastsurfer.sh
The Bug is actually in stools.sh:translate_cases, where the lines are not correct parsed.
This Commit fixes this issue and properly parses the file.
Also, the check to find non-existent files in stools.sh:check_subject_images.

This Commit also adds a check if files are symlinks, which may cause problems with folders mounted in singularity (this needed a small change in srun_fastsurfer.sh - an additional argument to check_subject_images).

dkuegler committed:
Fix --cpus-per-task in surface jobs.

dkuegler committed:
Bugfixing T2 processing in run_fastsurfer.sh

dkuegler commented:
Basic testing complete, now testing on a bigger dataset.

m-reuter commented:
looks generally ok. As long as the tests work and normal processing is not breaking, we can merge it.

dkuegler committed:
T1T2 registration needs a FreeSurfer license, make sure the license is passed for seg_only module in srun_fastsurfer.sh as well as add a license check if a T2 image is passed to run_fastsurfer.sh

dkuegler committed:
Fix singularity parameters for surface pipeline in srun_fastsurfer.sh

dkuegler committed:
Only check t2 file name, if we are running the segmentation pipeline
Pass SLURM_ARRAY environment variables through to the segmentation process (not doing so causes the segmentation process to not detect which "batch" to process)
Fix number of jobs in brun_fastsurfer.sh
Fix device detection in brun_fastsurfer.sh

dkuegler committed:
Fix typo in --reg_mode (instead of --regmode)
